src/genome.py

```diff
@@ -88,154 +97,260 @@ def __str__(self) -> str:
 class ListGenome(Genome):
     """
     Representation of a genome.
-
     Implements the Genome interface using Python's built-in lists
     """
 
     def __init__(self, n: int):
         """Create a new genome with length n."""
-        ...  # FIXME
+        #initialize the genome with no TE's yet.
+        self.nucleotide= ['-'] * n
+        self.TE = {}
+        self.active_TE = []
 
     def insert_te(self, pos: int, length: int) -> int:
         """
         Insert a new transposable element.
-
         Insert a new transposable element at position pos and len
         nucleotide forward.
-
         If the TE collides with an existing TE, i.e. genome[pos]
         already contains TEs, then that TE should be disabled and
         removed from the set of active TEs.
-
         Returns a new ID for the transposable element.
         """
-        ...  # FIXME
-        return -1
+
+        if len(self.TE) != 0:
+            TE_ID = max(self.TE) + 1
+        else:
+            TE_ID = 1
+        self.TE[TE_ID] = length
+        self.active_TE.append(TE_ID)
+        if self.nucleotide[pos] in self.active_TE:
+            self.disable_te(self.nucleotide[pos])
+        self.nucleotide[pos:pos] = [TE_ID]*length
+        return TE_ID
+
 
     def copy_te(self, te: int, offset: int) -> int | None:
         """
         Copy a transposable element.
-
-        Copy the transposable element te to an offset from its current
+        Copy the transposable element te to an offset from its current_node
         location.
-
         The offset can be positive or negative; if positive the te is copied
         upwards and if negative it is copied downwards. If the offset moves
         the copy left of index 0 or right of the largest index, it should
         wrap around, since the genome is circular.
-
         If te is not active, return None (and do not copy it).
         """
-        ...  # FIXME
+        if te not in self.active_TE:
+            return None
+        pos = self.nucleotide.index(te)
+        res = (pos + offset) % len(self.nucleotide)
+        return self.insert_te(res, self.TE[te])
+        
+            
 
     def disable_te(self, te: int) -> None:
         """
         Disable a TE.
-
         If te is an active TE, then make it inactive. Inactive
         TEs are already inactive, so there is no need to do anything
         for those.
         """
-        ...  # FIXME
+        self.active_TE.remove(te)
+        for i in range(len(self)):
+            if self.nucleotide[i] == te:
+                for j in range(self.TE[te]):
+                    self.nucleotide[i+j] = 'x'
+                break
```

Comment:
Ok, but you could also just remove the TE from `self.active_TE` and handle the string representation in `__str__`. There, if a nucleotide is an ID, and that ID is not in `self.active_TE` you can output the `'x'`. You don't need to spend linear time updating the genome here. But there is nothing wrong with this implementation; it is just possible to do it faster.